docs/design/2021-07-29-hidden-sysvars.md

```diff
@@ -0,0 +1,176 @@
+
+These are a subcategory of “experimental” feature flags, with the difference being that the feature is still in active development, and is not ready for users to try the experiment. There may be significant known pitfalls.
+
+Because refactoring code itself can introduce bugs, if the code for these features exists in the server, in the interest of transparency it should be disclosed to the DBA, even if it is highly recommended that the user never use this feature.
+
+**Recommended solution:**
+
+* These variables should be migrated to use `tidb_experimental_feature_switch`.
+* The sysvar documentation for `tidb_experimental_feature_switch` will not document these unsupported flags (See Appendix #1) but instead say:
+   > Your TiDB server installation may include additional experimental features not in this list. Enabling them is not recommended and could result in data loss.
+
+### #4 Non-boolean Experimental Options
+
+This category of variables is different from Experimental/In Development feature flags because the tidb_experimental_feature_switch only handles boolean switches.
+
+These variables almost always depend on an experimental feature flag (which will move to `tidb_experimental_feature_switch`). If they don’t, one might need to be created.
+
+**Recommended Solution:**
+
+* Because the system variable system supports dynamic registration/unregistration, these variables should be removed when the experiment is disabled, and added when the experiment is not enabled. 
+* In some cases these variables might have been miscategorized and are instead “Stable” but intended to be deprecated or advanced configuration variables (category #1).
+
+### Future Handling
+
+Internally, we need an explicit guidebook for how the variables are added and removed with the evolution of TiDB:
+
+* When and how a new variable is added. The cons/pros of adding a variable compared with a configuration item.
+* When and how a variable will be removed.
+
+Once it is decided, it’s a good idea to publish it the design docs.
+
+### Documentation Standards
+
+There are currently 75 undocumented system variables (not including noop sysvars). Because several hidden variables are actually already documented, the lack of documentation is really an orthogonal problem that also needs fixing.
+
+It is expected that all system variables will be documented, with the exception of category #3 which in the recommendation is explicitly “documented as not documented”.
+
+## Implementation Plan
+
+#### Appendix #1 - tidb_experimental_feature_switch documentation
+
+##### tidb_experimental_feature_switch
+
+* Scope: SESSION | GLOBAL
+* Default value: (system dependent)
+* This variable allows you to enable or disable TiDB Server features which have not reached GA maturity.
+Individual values can be changed by specifying a comma separated list. For example: `SET GLOBAL tidb_experimental_feature_switch = 'alter_placement=ON,cascades_planner=ON';`
```

Comment:
We will need  SHOW GLOBAL VARIABLES LIKE 'tidb_experimental_feature_switch' to see feature status?
What if we use SHOW FEATURE STATUS? Then feature status and description can be shown together for better understanding.

Reply:
As discussed, we can add `SHOW EXPERIMENTAL FEATURES` (or similar). My suggestion would just be that it is in addition to the other methods, since otherwise we would need to invent new syntax to update the variable as well, which becomes complicated.